keyboards/gmmk/numpad/keymaps/vial/keymap.c

```diff
@@ -0,0 +1,81 @@
+/* Copyright 2021 Glorious, LLC <salman@pcgamingrace.com>
+ * Modified 2022 by rustedaperture for qmk_firmware
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "analog.h"
+#include "qmk_midi.h"
+
+// clang-format off
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+//      NUM      /       *       -
+//      7        8       9       +
+//      4        5       6       ENCODER
+//      1        2       3       RET
+//      0                        .
+
+  [0] = LAYOUT(
+    MO(1),   KC_PSLS,   KC_PAST,   KC_PMNS,
+    KC_P7,   KC_P8,     KC_P9,     KC_PPLS,
+    KC_P4,   KC_P5,     KC_P6,     KC_CALC,
+    KC_P1,   KC_P2,     KC_P3,     KC_PENT,
+    KC_P0,                         KC_PDOT
+  ),
+  [1] = LAYOUT(
+    _______,   KC_PSLS,   KC_PAST,   KC_PMNS,
+    KC_P7,     RGB_VAI,   KC_P9,     KC_PPLS,
+    RGB_RMOD,  KC_P5,     RGB_MOD,   KC_CALC,
+    KC_P1,     RGB_VAD,   KC_P3,     KC_PENT,
+    RGB_TOG,                         QK_BOOT
+  ),
+  [2] = LAYOUT(
+    _______,   _______,   _______,   _______,
+    _______,   _______,   _______,   _______,
+    _______,   _______,   _______,   _______,
+    _______,   _______,   _______,   _______,
+    _______,                         _______
+  ),
+  [3] = LAYOUT(
+    _______,   _______,   _______,   _______,
+    _______,   _______,   _______,   _______,
+    _______,   _______,   _______,   _______,
+    _______,   _______,   _______,   _______,
+    _______,                         _______
+  ),
```

Comment:
```suggestion
  )
```
No need to hardcode filler `KC_TRNS` layers. Dynamic keymaps will automatically fill in "missing" layers at compile-time up to the maximum set in `dynamic_keymap.layer_count`. (By default, this is 4.)

Reply:
Aight, i just did what the docs said :) gonna put a change soon